file-formats/auth/type/zif_aff_auth_v1.intf.abap

```diff
@@ -0,0 +1,108 @@
+INTERFACE zif_aff_auth_v1
+  PUBLIC.
+  TYPES:
+    "! <p class="shorttext">ABAP Dictionary</p>
+    "! ABAP dictionary
+    "! $required
+    BEGIN OF ty_abap_dictionary,
+      "! <p class="shorttext">Data Element</p>
+      "! Name of the data element
+      "! $required
+      dtel                           TYPE c         LENGTH 30,
+      "! <p class="shorttext">Domain</p>
+      "! Name of the domain
+      domname                        TYPE c         LENGTH 30,
+      "! <p class="shorttext">Output Length</p>
+      "! Length of the data element
+      outputlen                      TYPE n         LENGTH 6,
+      "! <p class="shorttext">Conversion Routine</p>
+      "! Conversion routine
+      conversion_routine             TYPE c         LENGTH 5,
+      "! <p class="shorttext">Organizational Level</p>
+      "! Organizational level information
+      organizational_level           TYPE c         LENGTH 120,
+    END OF ty_abap_dictionary,
+
+    "! <p class="shorttext">Authorization Maintenance</p>
+    "! Authorization maintenance
+    BEGIN OF ty_maintenance_dialog,
+      "! <p class="shorttext">Maintenance Dialog</p>
+      "! Maintenance dialog
+      maintenance_dialog             TYPE string,
+      "! <p class="shorttext">Some Authorization Objects use an own Maintenance Dialog</p>
+      "! Some authorization objects use an own maintenance dialog
+      maintenance_exit_dialog_object TYPE abap_bool,
+    END OF ty_maintenance_dialog,
+
+    "! <p class="shorttext">Search Help</p>
+    "! Search help
+    "! $required
+    BEGIN OF ty_provide_search_help,
+      "! <p class="shorttext">Check Table</p>
+      "! Check table
+      table                          TYPE c         LENGTH 30,
+      "! <p class="shorttext">Search Help Type</p>
+      "! Search help type
+      search_help_type               TYPE string,
+      "! <p class="shorttext">Some Authorization Objects use an Object Field Search Help</p>
+      "! Some authorization objects use an object field search help
+      maintenance_exit_dialog_object TYPE abap_bool,
+    END OF ty_provide_search_help,
+
+    "! <p class="shorttext">Table Details</p>
+    "! Table
+    BEGIN OF ty_tablestruc,
+      "! <p class="shorttext">Class</p>
+      "! Authorization object class
+      "! $showAlways
+      oclss                          TYPE c         LENGTH 4,
+      "! <p class="shorttext">Object</p>
+      "! Authorization object
+      "! $showAlways
+      objct                          TYPE c         LENGTH 10,
+      "! <p class="shorttext">Description</p>
+      "! Description
+      "! $showAlways
+      ttext                          TYPE c         LENGTH 60,
+      "! <p class="shorttext">Search Help</p>
+      "! Search help
+      "! $showAlways
+      searchhelp                     TYPE c         LENGTH 30,
```

Comment:
I would suggest following name.

```suggestion
      search_help                     TYPE c         LENGTH 30,
```

Furthermore, would you also show the field search help, if no content is provided by the user or the field is empty? Otherwise, I would remove the annotation `$showAlways`.